docs/developerguide/Headerfiles.md

```diff
@@ -0,0 +1,60 @@
+# 3. Headers Files
+
+## a) Create a "Module.h" header file
+
+- The file **"Module.h"** defines the mandatory `MODULE_NAME`. Thunder provides a trace and warning reporting feature. To accurately identify the source of a warning, Thunder needs to know the human-readable name of the package (executable or library). This package name is defined by the `MODULE_NAME` and declared by the `MODULE_NAME_DECLARATION()`.
+
+- It also includes wpeframework headers "<plugins/plugins.h>".
+```C++
+#ifndef MODULE_NAME
+#define MODULE_NAME Plugin_IOController
+#endif
+
+#include <plugins/plugins.h>
+```
+Refer [FooPlugin Module.h](./FooPlugin/Module.h) header file.
+
+## b) Create a "Module.cpp" file
+- This file defines the mandatory `MODULE_NAME_DECLARATION` as mentioned below.
+```C++
+#include "Module.h"
+MODULE_NAME_DECLARATION(BUILD_REFERENCE)
+```
+Refer [FooPlugin Module.cpp](./FooPlugin/Module.cpp).
+
+## c) Create "FooPlugin.h" main header file
+- This main header file declares the Plugin implementation class.
+
+Refer [FooPlugin.h](./FooPlugin/FooPlugin.h) header file.
+
+### Highlighting important sections from the Header file:
+
+- RDK services are implemented as Thunder Plugins and must adhere to the **PluginHost::IPlugin** interface. If a RDK service handles WEB requests it implements **PluginHost::IWeb**. If it activates/deactivates and handles JSON-RPC it implements **PluginHost::IDispatcher (or derives from PluginHost::JSONRPC)**.
+```C++
```

Comment:
Can we ensure that the code blocks are horizontally aligned correctly on the Headers and Source Files pages and are of the same width. I see some of them are not left aligned correctly and seem to have different widths.

https://pavankumar-hm.github.io/rdkservices/#/developerguide/Headerfiles